framework/src/meshgenerators/BlockDeletionGenerator.C

```diff
@@ -22,18 +22,27 @@ BlockDeletionGenerator::validParams()
 
   params.addClassDescription(
       "Mesh generator which removes elements with the specified subdomain ID");
-  params.addRequiredParam<SubdomainID>("block_id", "The block to be deleted");
+  params.addParam<std::vector<SubdomainID>>("block_ids", "The list of blocks to be deleted");
+  params.addDeprecatedParam<SubdomainID>(
+      "block_id", 0, "The block to be deleted. Use block_ids instead");
 
   return params;
 }
 
 BlockDeletionGenerator::BlockDeletionGenerator(const InputParameters & parameters)
-  : ElementDeletionGeneratorBase(parameters), _block_id(getParam<SubdomainID>("block_id"))
+  : ElementDeletionGeneratorBase(parameters),
+    _block_ids(getParam<std::vector<SubdomainID>>("block_ids"))
 {
+  // Handle deprecated parameter
+  if (!parameters.isParamSetByUser("block_id"))
```

Comment:
Should be using `isParamValid()` in the cases where this is set by an action

Reply:
but param is always valid since there is a default for deprecated parameters?

Reply:
oh right I need to use isParamSetByAddParam

Reply:
It shouldn't have a default. I'm not sure why you set one

Reply:
Indeed it shouldnt but it's expecting 3 arguments.

Reply:
Your use of it is wrong:
https://mooseframework.inl.gov/docs/doxygen/moose/classInputParameters.html#ab9a4d5c47f1c714eb3469c1886720d44

name, doc_string, deprecation message

Reply:
nice, seems I cant read anymore